File: Solutions/10327/csharp/Program.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Quera {
    public static class Program {
        public static void Main() {
            var firstLine = Console.ReadLine().Split(' ');
            var numOfCodes = Convert.ToInt32(firstLine[0]);
            var validChars = firstLine[1].Distinct().ToList();

            for (var i = 0; i < numOfCodes; i++) {
                var codeChars = Console.ReadLine().Distinct().ToList();
                var isValid = IsEqual(codeChars, validChars);
                Console.WriteLine(isValid ? "Yes" : "No");
            }
        }

        private static bool IsEqual<T>(IReadOnlyCollection<T> a, IReadOnlyCollection<T> b) =>
            !a.Except(b).Any() && !b.Except(a).Any();
    }
}

File: Solutions/10327/python3/program.py
def main():
    num_of_codes, valid_code = input().split()
    valid_chars = set(valid_code)

    for i in range(int(num_of_codes)):
        code_chars = set(input())
        is_valid = code_chars == valid_chars
        if is_valid:
            print("Yes")
        else:
            print("No")


if __name__ == '__main__':
    main()
